cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -2465,4 +2465,22 @@ const char* elt_int32_utf8_utf8_utf8_utf8_utf8(
       return nullptr;
   }
 }
+
+FORCE_INLINE
+int32_t instr_utf8(const char* string, int32_t string_len, const char* substring,
+                   int32_t substring_len) {
+  if (substring_len == 0) {
+    return 1;
+  }
+
+  int32_t end_idx =
+      (string_len - substring_len) <= 0 ? string_len : string_len - substring_len;
+
+  for (int i = 0; i < end_idx; i++) {
+    if (string[i] == substring[0] && strncmp(string + i, substring, substring_len) == 0) {
```

Comment:
You cannot use strncmp() since this terminates if it sees a '\0' char. You should use memcmp() with the appropriate size

Reply:
The substring_len is ok now, the comparing was worng, memcmp is now ok